grpc.go

```diff
@@ -590,38 +667,28 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		return emptyResp, err
 	}
 
-	// Update libcontainer configuration for specific cases not handled
-	// by the specconv converter.
-	if err = a.updateContainerConfig(ociSpec, config, ctr); err != nil {
-		return emptyResp, err
-	}
-
-	containerPath := filepath.Join("/tmp/libcontainer", a.sandbox.id)
-	factory, err := libcontainer.New(containerPath, libcontainer.Cgroupfs)
-	if err != nil {
-		return emptyResp, err
-	}
-
-	ctr.container, err = factory.Create(req.ContainerId, config)
+	// In order to comply with the OCI specification, the spec must be
+	// written to a file named 'config.json' located in the bundle
+	// directory. https://github.com/opencontainers/runtime-spec
+	err = writeSpecToFile(ociSpec)
 	if err != nil {
 		return emptyResp, err
 	}
-	ctr.config = *config
 
-	ctr.initProcess, err = buildProcess(req.OCI.Process, req.ExecId)
+	// Change back to root dir
+	err = os.Chdir(oldcwd)
 	if err != nil {
 		return emptyResp, err
 	}
 
-	if err = a.execProcess(ctr, ctr.initProcess, true); err != nil {
-		return emptyResp, err
-	}
-
-	if err := a.updateSharedPidNs(ctr); err != nil {
+	// Update libcontainer configuration for specific cases not handled
+	// by the specconv converter.
+	if err = a.updateContainerConfig(ociSpec, config, ctr); err != nil {
 		return emptyResp, err
 	}
 
-	return emptyResp, a.postExecProcess(ctr, ctr.initProcess)
+	// reduce cyclomatic complexity of this function
```

Comment:
Please can you remove this comment - feel free to mention you needed to refactor to reduce cyclomatic complexity in the commit message though.